File: HealthAppMVC/SharedDto/DoctorDtos/DoctorDto.cs
namespace SharedDto.DoctorDtos
{
    public class DoctorDto
    {
        public int DoctorId { get; set; }

        public string FullName { get; set; }

        public string Specialisation { get; set; }

        public decimal ConsultationFee { get; set; }

        public int YearsOfExperience { get; set; }

        public bool IsActive { get; set; }

        public string DoctorPhoneNo { get; set; }

        public string DoctorEmail { get; set; }
    }
}

File: HealthApp.Angular/src/app/dtos/doctor.dto.ts
export interface DoctorDto {
  doctorId: number;
  fullName: string;
  specialisation: string;
  doctorPhoneNo: string;
  doctorEmail: string;
  yearsOfExperience: number;
  consultationFee: number;
  isActive: boolean;
}

export interface DoctorCreateDto {
  fullName: string;
  specialisation: string;
  doctorPhoneNo: string;
  doctorEmail: string;
  yearsOfExperience: number;
  consultationFee: number;
}

export interface DoctorUpdateDto {
  fullName: string;
  specialisation: string;
  doctorPhoneNo: string;
  doctorEmail: string;
  yearsOfExperience: number;
  consultationFee: number;
  isActive: boolean;
}

export interface SpecialisationOption {
  label: string;
  value: string;
}

export interface DoctorStatusUpdateResponse {
  message: string;
}

export type DoctorAvailabilitySlotStatus =
  | 'Available'
  | 'Booked'
  | 'DoctorOnLeave';

export interface DoctorAvailabilitySlotDto {
  timeSlot: string;
  isAvailable: boolean;
  status: DoctorAvailabilitySlotStatus;
}

export interface DoctorAvailabilityDto {
  doctorId: number;
  date: string;
  isDoctorOnLeave: boolean;
  message: string;
  slots: DoctorAvailabilitySlotDto[];
}